tests/end2end/test_nic.py

```diff
@@ -241,3 +247,145 @@ def test_nic_crud(dpm_mode_cpcs):  # noqa: F811
                 part.delete()
             if adapter:
                 adapter.delete()
+
+
+def test_nic_backing_port_port_based(dpm_mode_cpcs):  # noqa: F811
+    # pylint: disable=redefined-outer-name
+    """
+    Test Nic.backing_port() for port-based NICs (e.g. RoCE, CNA).
+    """
+    if not dpm_mode_cpcs:
+        pytest.skip("HMC definition does not include any CPCs in DPM mode")
+
+    logger = setup_logging(LOGGING, 'test_nic_backing_port_port_based',
+                           LOG_FILE)
+    logger.debug("Entered test function")
+
+    cpc = random.choice(dpm_mode_cpcs)
+    logger.debug("Testing with CPC %s", cpc.name)
+    client = cpc.manager.client
+
+    port_nics = []
+    partitions = cpc.partitions.list()
+    for partition in random.sample(partitions, k=len(partitions)):
```

Comment:
I don't get why you iterate over all partitions, but random order? Isn't it "random" anyway already?

Reply:
I want to avoid iterating through all NICs of all partitions, just to pick one for the test. So I am limiting the iteration of NICs until we have 5 (or less if the loop exhausts before that). In order not to use the same partitions over and over again, I am randomizing the order in which the partitions are iterated, so we get the 5 NICs from potentially any partition.